packages/integration-sdk-testing/src/jest.ts

```diff
@@ -9,25 +9,141 @@ import {
+       * const stepResult = await executeStepWithDependencies({
+       *   stepId: Steps.FETCH_USERS.id,
+       *   invocationConfig,
+       *   instanceConfig,
+       * });
+       *
+       * expect(stepResult).toMatchStepMetadata({
+       *   stepId: Steps.FETCH_USERS.id,
+       *   invocationConfig,
+       * });
+       * ```
+       */
+      toMatchStepMetadata(testConfig: StepTestConfig): R;
+
+      /**
+       * Used to verify that a collection of Entities matches the _type, _class,
+       * and schema defined for the collection, as well as any additional schema
+       * defined for the _class in the @jupiterone/data-model project
+       *
+       * @example
+       * ```ts
+       * const { collectedEntities } = await executeStepWithDependencies({
+       *   stepId: Steps.FETCH_USERS.id,
+       *   invocationConfig,
+       *   instanceConfig,
+       * });
+       *
+       * expect(collectedEntities).toMatchGraphObjectSchema({
+       *   _class: 'User',
+       *   _type: 'acme_user',
+       *   schema: {
+       *     properties: {
+       *       admin: { type: 'boolean' }
+       *     }
+       *   }
+       * })
+       * ```
+       */
+      toMatchGraphObjectSchema(params: ToMatchGraphObjectSchemaParams): R;
+
+      /**
+       * Used to verify that a collection of Direct Relationships matches the
+       * _type, _class, and schema defined for the collection, as well as any
+       * additional schema common to _all_ direct relationships.
+       *
+       * @example
+       * ```ts
+       * const { collectedRelationships } = await executeStepWithDependencies({
+       *   stepId: Steps.FETCH_USERS.id,
+       *   invocationConfig,
+       *   instanceConfig,
+       * });
+       *
+       * expect(collectedRelationships).toMatchGraphObjectSchema({
+       *   _class: RelationshipClass.HAS,
+       *   _type: 'acme_account_has_user',
+       *   schema: {
+       *     properties: {
+       *       displayName: { const: 'HAS' }
+       *     }
+       *   }
+       * })
+       * ```
+       */
+      toMatchDirectRelationshipSchema(
         params?: ToMatchRelationshipSchemaParams,
       ): R;
 
-      toTargetEntities(entities: Entity[]): R;
+      /**
+       * Used to verify that a collection of Mapped Relationships is able to
+       * create actual relationships to a set of Entities by matching on the
+       * `targetEntity` and `targetFilterKeys` properties.
+       *
+       * @example
+       * ```ts
+       * const {
+       *   collectedRelationships: firewallRuleMappedRelationships,
+       * } = await executeStepWithDependencies({
+       *   stepId: Steps.FETCH_USERS.id,
+       *   invocationConfig,
+       *   instanceConfig,
+       * });
+       *
+       * const globalInternetEntity = buildGlobalInternetEntity();
+       *
+       * expect(firewallRuleMappedRelationships).toTargetEntities(
+       *   [globalInternetEntity],
+       * )
+       * ```
+       */
+      toTargetEntities(
```

Comment:
The other jest matchers contain the word `match` in them. Should we update this function to: `toMatchTargetEntities`?

Reply:
Would be interesting to discuss the name here.

Reply:
Here's a usage https://github.com/JupiterOne/graph-aws/blob/main/src/steps/services/sqs/executionHandlers/queues.test.ts#L95

```ts
```

Reply:
See conversation from the original PR https://github.com/JupiterOne/sdk/pull/476#discussion_r647047907

We decided to rename from `toCreateValidRelationshipsToEntities` -> `toTargetEntities`